### PR TITLE
BUG: Fix windowing over read-only arrays

### LINKED ISSUE #27766: Windowing fails on read-only arrays
#### Code Sample, a copy-pastable example if possible

```python
>>> arr = np.array([1, 3, np.nan, 3, 5])
>>> arr.setflags(write=False)
>>> pd.Series(arr).rolling(2).mean().values
---------------------------------------------------------------------------
ValueError: assignment destination is read-only
```

#### Expected Output
```py
>>> pd.Series(arr).rolling(2).mean().values
array([nan,  2., nan, nan,  4.])
```
#### Problem description

https://github.com/pandas-dev/pandas/pull/27391#discussion_r309941141 missed a copy call before writing into array, which is user's data and which may be read-only.

#### Output of ``pd.show_versions()``

pandas 0.25.0

### PR DESCRIPTION
**Pull request by kernc**
- [x] closes #27766
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by kernc** [comment]
https://github.com/pandas-dev/pandas/pull/27736, which touches these files, is scheduled for 1.0. I'd prefer to see this in 0.25.1. :grin:

File: pandas/core/window.py [hunk 1]
```diff
@@ -246,8 +246,11 @@ def _prep_values(self, values: Optional[np.ndarray] = None) -> np.ndarray:
             except (ValueError, TypeError):
                 raise TypeError("cannot handle this type -> {0}".format(values.dtype))
 
-        # Always convert inf to nan
-        values[np.isinf(values)] = np.NaN
+        # Convert inf to nan for C funcs
+        inf = np.isinf(values)
+        if inf.any():
+            values = values.copy()  # GH-27766, Don't write into user's data
```
Comment (jbrockmendel) [inline pandas/core/window.py]: comment on previous line.

is the issue that we dont want to overwrite user's data, or that the user might pass read-only array?

Comment (kernc) [reply pandas/core/window.py]: Mmm, actually yeah, it's the first.

File: pandas/tests/window/test_rolling.py [hunk 2]
```diff
@@ -326,3 +326,10 @@ def test_rolling_axis_count(self, axis_frame):
 
         result = df.rolling(2, axis=axis_frame).count()
         tm.assert_frame_equal(result, expected)
+
+    def test_readonly_array(self):
+        arr = np.array([1, 3, np.nan, 3, 5])
```
Comment (WillAyd) [inline pandas/tests/window/test_rolling.py]: Can you add a ref to the GH number as a comment here?

File: pandas/tests/window/test_rolling.py [hunk 3]
```diff
@@ -326,3 +326,10 @@ def test_rolling_axis_count(self, axis_frame):
...
+        arr.setflags(write=False)
+        tm.assert_series_equal(
+            pd.Series(arr).rolling(2).mean(), pd.Series([np.nan, 2, np.nan, np.nan, 4])
```
Comment (WillAyd) [inline pandas/tests/window/test_rolling.py]: Just assign the first argument to `result` and the second to `expected` and do tm.assert_series_equal(result, expected) (standard in tests)

**Review by WillAyd — CHANGES_REQUESTED** [review]
minor nits on tests otherwise lgtm

**Review by WillAyd — APPROVED** [review]
lgtm @mroeschke

**Comment by jreback** [comment]
thanks @kernc 

very nice putting up a bug report AND a PR!

**Pull request merged**